ops/log.py

```diff
@@ -87,5 +99,220 @@ def except_hook(etype: type[BaseException], value: BaseException, tb: types.Trac
+    """
+
+    INPUT_VALIDATION_FAIL = 'input_validation_fail'
+
+
+class _SecurityEventMalicious(enum.Enum):
+    """Security event names for malicious events.
+
+    See https://cheatsheetseries.owasp.org/cheatsheets/Logging_Vocabulary_Cheat_Sheet.html
+    """
+
+    MALICIOUS_EXCESS_404 = 'malicious_excess_404'
+    MALICIOUS_EXTRANEOUS = 'malicious_extraneous'
+    MALICIOUS_ATTACK_TOOL = 'malicious_attack_tool'
+    MALICIOUS_CORS = 'malicious_cors'
+    MALICIOUS_DIRECT_REFERENCE = 'malicious_direct_reference'
+
+
+class _SecurityEventPrivilege(enum.Enum):
+    """Security event names for privilege and permissions events.
+
+    See https://cheatsheetseries.owasp.org/cheatsheets/Logging_Vocabulary_Cheat_Sheet.html
+    """
+
+    PRIVILEGE_PERMISSIONS_CHANGED = 'privilege_permissions_changed'
+
+
+class _SecurityEventSensitive(enum.Enum):
+    """Security event names for sensitive data events.
+
+    See https://cheatsheetseries.owasp.org/cheatsheets/Logging_Vocabulary_Cheat_Sheet.html
+    """
+
+    SENSITIVE_CREATE = 'sensitive_create'
+    SENSITIVE_READ = 'sensitive_read'
+    SENSITIVE_UPDATE = 'sensitive_update'
+    SENSITIVE_DELETE = 'sensitive_delete'
+
+
+class _SecurityEventSequence(enum.Enum):
+    """Security event names for sequence events.
+
+    See https://cheatsheetseries.owasp.org/cheatsheets/Logging_Vocabulary_Cheat_Sheet.html
+    """
+
+    SEQUENCE_FAIL = 'sequence_fail'
+
+
+class _SecurityEventSession(enum.Enum):
+    """Security event names for session events.
+
+    See https://cheatsheetseries.owasp.org/cheatsheets/Logging_Vocabulary_Cheat_Sheet.html
+    """
+
+    SESSION_CREATED = 'session_created'
+    SESSION_RENEWED = 'session_renewed'
+    SESSION_EXPIRED = 'session_expired'
+    SESSION_USE_AFTER_EXPIRE = 'session_use_after_expire'
+
+
+class _SecurityEventSystem(enum.Enum):
+    """Security event names for system events.
+
+    See https://cheatsheetseries.owasp.org/cheatsheets/Logging_Vocabulary_Cheat_Sheet.html
+    """
+
+    SYS_STARTUP = 'sys_startup'
+    SYS_SHUTDOWN = 'sys_shutdown'
+    SYS_RESTART = 'sys_restart'
+    SYS_CRASH = 'sys_crash'
+    SYS_MONITOR_DISABLED = 'sys_monitor_disabled'
+    SYS_MONITOR_ENABLED = 'sys_monitor_enabled'
+
+
+class _SecurityEventUser(enum.Enum):
+    """Security event names for user events.
+
+    See https://cheatsheetseries.owasp.org/cheatsheets/Logging_Vocabulary_Cheat_Sheet.html
+    """
+
+    USER_CREATED = 'user_created'
+    USER_UPDATED = 'user_updated'
+    USER_ARCHIVED = 'user_archived'
+    USER_DELETED = 'user_deleted'
+
+
+_SecurityEvent = typing.Union[
+    _SecurityEventAuthN,
+    _SecurityEventAuthZ,
+    _SecurityEventCrypt,
+    _SecurityEventExcess,
+    _SecurityEventUpload,
+    _SecurityEventInput,
+    _SecurityEventMalicious,
+    _SecurityEventPrivilege,
+    _SecurityEventSensitive,
+    _SecurityEventSequence,
+    _SecurityEventSession,
+    _SecurityEventSystem,
+    _SecurityEventUser,
+]
+
+
+def _log_security_event(
```

Comment:
TBD if this ends up being `log_security_event` and added to the `ops` namespace. So far, I have one vote in favour and silence otherwise.